subprojects/dependency-management/src/testFixtures/groovy/org/gradle/integtests/fixtures/publish/ModuleSpec.groovy

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.fixtures.publish
+
+import org.gradle.test.fixtures.server.http.MavenHttpRepository
+
+class ModuleSpec {
+    private final String groupId
+    private final String artifactId
+    private final Map<String, ModuleVersionSpec> versions = [:].withDefault { new ModuleVersionSpec(groupId, artifactId, it) }
+
+    private boolean expectGetMetatada
+
+    ModuleSpec(String group, String name) {
+        groupId = group
+        artifactId = name
+    }
+
+    void version(String version, @DelegatesTo(value=ModuleVersionSpec, strategy = Closure.DELEGATE_ONLY) Closure<Void> versionSpec = {}) {
+        versionSpec.delegate = versions[version]
+        versionSpec.resolveStrategy = Closure.DELEGATE_ONLY
+        versionSpec()
+    }
+
+    void publish(MavenHttpRepository repository) {
```

Comment:
To clarify my above comment: we aren't really 'publishing' here, as in we are not using the Gradle publish architecture. I'm OK with this method name, but we need to find a good way to differentiate between the fixtures when we use them to verify the details of a publication, and the fixtures when we use the to _produce_ a publication for tests to consume.

Reply:
I have renamed the method to `produce`

Reply:
I would choose 'build', but I guess this doesn't really impact the use of the new DSL in any way.